core/tests/protractor_desktop/explorationImprovementsTab.js

```diff
@@ -100,30 +98,46 @@ describe('Answer Details Improvements', function() {
     explorationEditorMainTab.setInteraction('EndExploration');
     explorationEditorPage.saveChanges();
     workflow.publishExploration();
-  });
-
-  it('checks solicit answer details feature', function() {
-    users.login('learner@user.com');
-    libraryPage.get();
-    libraryPage.findExploration(EXPLORATION_TITLE);
-    libraryPage.playExploration(EXPLORATION_TITLE);
-    explorationPlayerPage.submitAnswer('TextInput', 'One');
-    explorationPlayerPage.submitAnswerDetails('I liked this choice of answer');
-    explorationPlayerPage.expectExplorationToNotBeOver();
-
-    oppiaLogo.click();
-    general.acceptAlert();
     users.logout();
-    users.login('creator@user.com');
-    creatorDashboardPage.get();
-    creatorDashboardPage.navigateToExplorationEditor();
-    explorationEditorPage.navigateToImprovementsTab();
+  });
 
-    var task = improvementsTab.getAnswerDetailsTask('One');
-    improvementsTab.clickTaskActionButton(task, 'Review Answer Details');
-    improvementsTab.verifyAnswerDetails('I liked this choi...', 1);
-    improvementsTab.closeModal();
-    users.logout();
+  describe('Solicit answer details', function() {
+    beforeAll(function() {
+      users.login('learner@ExplorationAnswerDetails.com');
+      libraryPage.get();
+      libraryPage.findExploration(EXPLORATION_TITLE);
+      libraryPage.playExploration(EXPLORATION_TITLE);
+      explorationPlayerPage.submitAnswer('TextInput', 'One');
+      explorationPlayerPage.submitAnswerDetails(
+        'I liked this choice of answer');
+      explorationPlayerPage.expectExplorationToNotBeOver();
+      oppiaLogo.click();
+      general.acceptAlert();
+      users.logout();
+    });
+
+    it('is visible for creators', function() {
+      users.login('creator@ExplorationAnswerDetails.com');
+      creatorDashboardPage.get();
+      creatorDashboardPage.navigateToExplorationEditor();
+      explorationEditorPage.navigateToImprovementsTab();
+
+      var task = improvementsTab.getAnswerDetailsTask('One');
+      improvementsTab.clickTaskActionButton(task, 'Review Answer Details');
+      improvementsTab.verifyAnswerDetails('I liked this choi...', 1);
+      improvementsTab.closeModal();
+      users.logout();
+    });
+
+    it('is visible for guests', function() {
+      general.openEditor(this.expId);
+      explorationEditorPage.navigateToImprovementsTab();
+
+      var task = improvementsTab.getAnswerDetailsTask('One');
+      improvementsTab.clickTaskActionButton(task, 'Review Answer Details');
+      improvementsTab.verifyAnswerDetails('I liked this choi...', 1);
```

Comment:
Perhaps add checks to verify that the various buttons/input fields do not exist.

Reply:
Done